tests/charts/api_tests.py

```diff
@@ -686,13 +793,19 @@ def test_update_chart_validate_datasource(self):
         db.session.delete(chart)
```

Comment:
Perhaps this should be in a `finally` or a teardown?

Reply:
Do you mean `finally` in `try` block? I think `tearDown` would be a a bit hacky because it is test specific object. It is more general problem. There are many tests that create some specific objects and delete them after assertion. I wonder how should we refactor them. Creating tests specific fixtures or context managers would result in many additional lines of code and make tests less readable. On the other hand if any test fails that means something should be fixed and next failures caused by side effects of the failed test may be not so important.

Reply:
I was thinking `finally` in a `try` block, yes, in an attempt to isolate test failures. Cascading failures make the test suite much more challenging to work with on a day to day basis - lots to dig through to figure out the issue. I'm not sure what the best refactor here is to ensure cleanup without bloating the code.